### ISSUE TITLE
Weird issue in epoll module

### ISSUE THREAD

**Issue by AlexDaniel** [rakudo/rakudo#3244]
I haven't checked what's going on there yet, but filing a ticket so that we don't forget about it.

* [epoll](https://modules.perl6.org/dist/epoll) – Fail, Bisected: 7e76762c52eef144c6f0cbf8970441aad43d91ad

```
===> Testing: epoll:ver<0.3>
t/00-meta.t .... ok
Use of uninitialized value of type Any in numeric context
  in submethod TWEAK at /root/rakudo/Blin/data/zef-data/store/epoll-0.3.tar.gz/epoll-0.3/lib/epoll.pm6 (epoll) line 57
Cannot unbox a type object (Any) to int.

# You planned 6 tests, but ran 1
t/01-simple.t .. 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 5/6 subtests 

Test Summary Report
-------------------
t/01-simple.t (Wstat: 65280 Tests: 1 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 6 tests but ran 1.
Files=2, Tests=2, 14 wallclock secs ( 0.06 usr  0.01 sys + 12.47 cusr  0.61 csys = 13.15 CPU)
Result: FAIL
```

**Comment by Kaiepi** [comment]
This doesn't appear to be related to the commits included with 7e76762. I double-checked to make sure I didn't overwrite anything in MoarVM by accident that could be responsible for this, but that doesn't seem to be the case. This is likely another NativeCall-related issue.

**Comment by niner** [comment]
This is interesting: put `...` into the bodies of the native subs and it works again:
`sub epoll_create1(int32 --> int32) is native { ... }`
`???` and `!!!` work, too, but `*` doesn't.

**Comment by niner** [comment]
The other way to make it work again is even weirder: `sub epoll_create1(int32 --> int32) is native(Str) {};` i.e. explicitly giving a Str to the `native` trait.
That's especially weird since the trait turns the Bool into exactly that: `$r does NativeCall::Native[$r, $native === True ?? Str !! $native];`
But at least `is native(True)` also makes it run. So it's really just the `is native` with a body that doesn't contain one of the stub code markers that triggers this.

It also seems to require the arguments to be native. Giving it a Pointer (which should also just be an int) makes the issue go away as well.

**Comment by niner** [comment]
Fixed in commit d662912e21fdad736900166f3d15b428f22af98b
